### PR TITLE
Avoid including sys-debug.so and libjulia-debug.so in the binary-dist.

### LINKED ISSUE #27753: Stop bundling debug versions of libraries
On 0.7-alpha, out of the 600M, 250M are just the debug libs. I am pretty sure that 99% of the users do not use the debug libs in any way. Would be nice to slim down our download.

### PR DESCRIPTION
**Pull request by ViralBShah**
Fixes #27753

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -357,7 +358,7 @@ endif
 	cp -R -L $(build_includedir)/julia/* $(DESTDIR)$(includedir)/julia
 	# Copy system image
 	$(INSTALL_M) $(build_private_libdir)/sys.$(SHLIB_EXT) $(DESTDIR)$(private_libdir)
-	$(INSTALL_M) $(build_private_libdir)/sys-debug.$(SHLIB_EXT) $(DESTDIR)$(private_libdir)
+#	$(INSTALL_M) $(build_private_libdir)/sys-debug.$(SHLIB_EXT) $(DESTDIR)$(private_libdir)
```
Comment (nalimilan) [inline Makefile]: It's weird to keep this but commented out. Better check whether `julia-debug` appears in `JL_LIBS`.

Comment (ViralBShah) [reply Makefile]: That's a good idea.

**Comment by ViralBShah** [comment]
One thing we can do is provide the debug libraries as an add-on in a separate download, rather than in the default.

**Comment by ViralBShah** [comment]
The `.dSYM` files on Mac are an additional 140M (on disk after installation).

**Comment by ViralBShah** [comment]
Triage should discuss whether we can go ahead with this.

**Comment by ViralBShah** [comment]
For the separate download, considerable more work will be need to package these things up, and adjust all the various bots and infrastructure.

**Comment by StefanKarpinski** [comment]
As far as I'm aware, we've never gotten a bug report based on the bundled debug version.

**Comment by StefanKarpinski** [comment]
Triage agrees: remove it.

**Comment by ViralBShah** [comment]
mac is failing on `cmdlineargs`. 

linux32 failed because it could not download the certificates (connection error) - so shouldn't be an issue.

**Pull request merged**